src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/ObjectWriter.cs

```diff
@@ -446,6 +446,23 @@ private void EmitObject(string objectFilePath, IReadOnlyCollection<DependencyNod
                         sectionWriter.Position,
                         nodeContents.Data,
                         nodeContents.Relocs));
+
+#if DEBUG
+                    // Pointer relocs should be aligned at pointer boundaries within the image.
+                    // Processing misaligned relocs (especially relocs that straddle page boundaries) can be
+                    // expensive on Windows.
+                    bool hasPointerRelocs = false;
+                    foreach (Relocation reloc in nodeContents.Relocs)
+                    {
+                        if ((reloc.RelocType is RelocType.IMAGE_REL_BASED_DIR64 && _nodeFactory.Target.PointerSize == 8) ||
+                            (reloc.RelocType is RelocType.IMAGE_REL_BASED_HIGHLOW && _nodeFactory.Target.PointerSize == 4))
+                        {
+                            hasPointerRelocs = true;
+                            Debug.Assert(reloc.Offset % _nodeFactory.Target.PointerSize == 0);
+                        }
```

Comment:
The stack trace metadata node has unaligned pointer relocs. Everything else is (AFAIK) aligned.